### PR TITLE
improved simplification of scalar factor for MatrixSymbol

### LINKED ISSUE #13490: missing symplify for MatrixSymbol with neg
following example failed to simplify `S.NegativeOne * S.NegativeOne`
```python
n, m = symbols('n m')
A = MatrixSymbol('A', n, m)
assert(A == = simplify(-(-A)) )
```

### PR DESCRIPTION
**Pull request by vruge**
-  some simplification were skipped for scalar coeff

example
```python
  n, m = symbols('n m')
  A = MatrixSymbol('A', n, m)

  assert(A == -(-A))
```

- added test

"Fixes #13490"

### CODE REVIEW

**Comment by ylemkimon** [comment]
It seems it’s not typo. result is rm_id-ed mmul, and should be compared to mmul.

**Comment by vruge** [comment]
@ylemkimon thank's

File: sympy/matrices/expressions/matexpr.py [hunk 1]
```diff
@@ -457,20 +457,13 @@ class Identity(MatrixExpr):
...
-
-    @property
-    def rows(self):
-        return self.args[0]
-
-    @property
-    def cols(self):
-        return self.args[0]
+    def __new__(cls, n, m = None):
+        if m is None: m = n
```
Comment (smichr) [inline sympy/matrices/expressions/matexpr.py]: -> `m = n if m is None else m`

File: sympy/matrices/expressions/matexpr.py [hunk 2]
```diff
@@ -457,20 +457,13 @@ class Identity(MatrixExpr):
...
+        return super(Identity, cls).__new__(cls, sympify(n), simplify(m))
```
Comment (smichr) [inline sympy/matrices/expressions/matexpr.py]: Why are you adding another argument? (And if it is necessary -- and I"m not convinced it is -- it should be `sympify` not `simplify`.)

Comment (smichr) [reply sympy/matrices/expressions/matexpr.py]: Identity matrix should be square, so you shouldn't be adding another argument, right?

Comment (vruge) [reply sympy/matrices/expressions/matexpr.py]: right, thank's :)

I swapped something with function eye e.g. in matlab which allowed eye(n,m)

**Comment by czgdp1807** [comment]
The issue associated with this PR has been fixed in the development version of sympy. Hence, closing this PR.

**Comment by oscarbenjamin** [comment]
@vruge thanks you for submitting this PR. I'm not sure exactly what happened but it seems that this was fixed somewhere else.

I don't know if this PR fixes anything else that isn't yet fixed on master - the tests added here are for things that are already fixed. If there are still unfixed problems on master then please open a new issue. That would likely need a new PR based on current master though so it is best to close this for now.

**Pull request closed without merging**